packages/expect-type/src/__tests__/index.test.ts

```diff
@@ -195,6 +195,60 @@ test('You can also check type guards & type assertions', () => {
   expectTypeOf(isString).guards.toBeString()
 })
 
+test('You can also check the `this` argument of functions using `.thisParam', () => {
+  type NoThisParam = (a: number) => void
+  type DateThisParam = (this: Date, a: number) => void
+  type UndefinedThisParam = (this: undefined, a: number) => void
+  type UnknownThisParam = (this: unknown, a: number) => void
+  type AnyThisParam = (this: any, a: number) => void
+
+  expectTypeOf<NoThisParam>().thisParam.toBeUnknown();
+  expectTypeOf<DateThisParam>().thisParam.toEqualTypeOf<Date>();
+  expectTypeOf<UndefinedThisParam>().thisParam.toBeUndefined();
+  expectTypeOf<UnknownThisParam>().thisParam.toBeUnknown();
+  expectTypeOf<AnyThisParam>().thisParam.toBeAny();
+})
+
+test('Of course, `.toEqualTypeOf` also distinguishes the `this` argument between functions', () => {
```

Comment:
```suggestion
test('`.toEqualTypeOf` also distinguishes the `this` argument between functions', () => {
```

I also wonder whether this test is stretching a bit to be part of the docs. If I were someone who used `this`, I would probably just assume it worked (which I suppose is why you added ‘of course’)!

Reply:
Is it possible to make a test that doesn't get included in the docs?